MapboxDirections/MBApproachType.swift

```diff
@@ -0,0 +1,39 @@
+import Foundation
+
+/**
+ Indicate how a route considers from which side of the road to approach a waypoint.
+ */
+@objc(MBApproachType)
+public enum ApproachType: UInt, CustomStringConvertible {
```

Comment:
“Type” is such an overloaded word that we should try very hard not to use it in any our Swift libraries’ public APIs. Some options:

* For the time being, the only two options imply that we could replace this enumeration with a simple `avoidsRoadCrossingOnArrival` or `allowsArrivingOnOppositeSide` flag.
* Call the enumeration `ApproachRestriction` and rename `unrestricted` to `none`.